staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,408 @@
+		existingFlag := foundAliasCmd.Flag(fl.Name)
+		if existingFlag == nil {
+			return args, fmt.Errorf("invalid alias flag %s in alias %s", fl.Name, args[0])
+		}
+		if searchInArgs(existingFlag.Name, existingFlag.Shorthand, foundAliasCmd.Flags(), args) {
+			// Don't modify the value implicitly, if it is passed in args explicitly
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
+	// Cobra (command.go#L1078) appends only root command's args into the actual args and ignores the others.
+	// We are appending the additional args defined in kuberc in here and
+	// expect that it will be passed along to the actual command.
+	rootCmd.SetArgs(args[1:])
+	return args, nil
+}
+
+// DefaultGetPreferences returns KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
+// where kubeconfig's defaults are residing in.
+func DefaultGetPreferences(kuberc string) (*config.Preference, error) {
+	kubeRCFile := RecommendedKubeRCFile
+	explicitly := false
+	if kuberc != "" {
+		kubeRCFile = kuberc
+		explicitly = true
+	}
+
+	if kubeRCFile == "" && os.Getenv("KUBERC") != "" {
+		kubeRCFile = os.Getenv("KUBERC")
+		explicitly = true
+	}
+
+	kubeRCBytes, err := os.ReadFile(kubeRCFile)
+	if err != nil {
+		if os.IsNotExist(err) && !explicitly {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	obj, gvk, err := codecs.UniversalDecoder().Decode(kubeRCBytes, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not be decoded gvk %s, err: %w", gvk, err)
+	}
+
+	expectedGK := schema.GroupKind{
+		Group: config.SchemeGroupVersion.Group,
+		Kind:  "Preference",
+	}
+	if gvk.GroupKind() != expectedGK {
+		return nil, fmt.Errorf("unsupported preference GVK %s", gvk.GroupKind().String())
+	}
+
+	var pref config.Preference
+	if err := scheme.Convert(obj, &pref, nil); err != nil {
+		return nil, fmt.Errorf("could not be decoded gvk %s, err: %w", gvk, err)
+	}
+
+	return &pref, nil
+}
+
+// Normally, we should extract this value directly from kuberc flag.
+// However, flag values are set during the command execution and
+// we are in very early stages to prepare commands prior to execute them.
+// Besides, we only need kuberc flag value in this stage.
+func getExplicitKuberc(args []string) string {
+	var kubercPath string
+	for i, arg := range args {
+		if arg == "--" {
+			// flags after "--" does not represent any flag of
+			// the command. We should short cut the iteration in here.
+			break
+		}
+		if arg == "--kuberc" {
+			if i+1 < len(args) {
```

Comment:
if they set `--kuberc` without a following arg, is that an error?

Reply:
Cobra catches this and errors out but it is better if we catch this and return error. Updated and added tests cases for the expected behavior.